### ISSUE TITLE
Using one navigation building logic instead of two (INavigationProvider and IMenuProvider)

### ISSUE THREAD

**Issue by andrasarva** [OrchardCMS/Orchard#5991]
I tried to make the BuildMenu(string) as the last comment was written by Sebastien Ros in this issue (https://github.com/OrchardCMS/Orchard/issues/5377) that use Merge() and Arrange() methods. The Merge() works well, but the Arrange() method ruins the hierarchy.
The Arrange method builds the hierarchy by splitting the position string at '.' characters so it's not working in INavigationProviders where the hierarchy is built differently.
For example Layouts menu has "8.5" position, so if I call the Arrange() method in BuildMenu(string) it will appear under the Task menu as a submenu.
Based on the idea described in the issue I linked above and based on the different hierarchy building logic it would be really recommended to refactor these implementations into one: Use INavigationProvider only for building menus in Orchard.

**Comment by sebastienros** [comment]
FYI it's a mess because something was working but not sufficient, and we needed to improve it a lot without breaking existing websites.

**Comment by sebastienros** [comment]
I checked, the issue is fixed and is not calling Arrange, like you are suggesting.

**Comment by Piedone** [comment]
Yes, because that was fixed by @andrasarva as part of #5377. This is not what this issue is about.
